packages/sdk/src/resolve/resolve-accounts.js

```diff
@@ -80,11 +84,33 @@ async function collectAccounts(ix, accounts, last, depth = 3) {
           Array.isArray(curr) ? [...prev, ...curr] : [...prev, curr],
         []
       )
+
+    // complete (flatmap) group replacement
+    // treat payer like authorizations, there can be multiple keys
+    ix.payer = Array.isArray(ix.payer) ? ix.payer
+    .map(d => (d === last.tempId ? payers : d))
+    .reduce(
+      (prev, curr) =>
+        Array.isArray(curr) ? [...prev, ...curr] : [...prev, curr],
+      []    
+    ) : [ix.payer]
   }
 }
 
 export async function resolveAccounts(ix) {
   if (isTransaction(ix)) {
+    if (!Array.isArray(ix.payer)) {
+      console.warn(
+        `
+        %cFCL Warning
+        ============================
+        "ix.payer" must be an array. Support for ix.payer as a singular is deprecated,
```

Comment:
Do you actually want to deprecate it or just leave both supported?

Reply:
My goal was to maintain backwards compatibility. I ran into an issue in `prepAccount`. Since there is a recursive nature to processing Authz method/object, there is no way of knowing if there are multiple "payer" keys needed in the interaction. 

Catch-22 here:
We decided to add the warning and process the interaction as an internal object. External projects need to no directly mess with the interaction object.

Reply:
Could phase this change out in multiple releases. Currently we only need to support multiple keys for payer for burning tokens for ledger. 

Could push out fcl-js release with the warning only
then in later release force interaction payer to be array.